packages/server/jiraImagesHandler.ts

```diff
@@ -0,0 +1,37 @@
+import {HttpRequest, HttpResponse} from 'uWebSockets.js'
+import uWSAsyncHandler from './graphql/uWSAsyncHandler'
+import {NO_IMAGE_BUFFER, JIRA_IMAGES_ENDPOINT} from './utils/atlassian/jiraImages'
+import getRedis from './utils/getRedis'
+
+const WAIT_MS = 500
+
+function wait(delayMS: number) {
+  return new Promise((resolve) => setTimeout(resolve, delayMS))
+}
+
+const getImageFromCache = async (fileName: string) => {
+  const redis = getRedis()
+  let imageBuffer = await redis.getBuffer(fileName)
+  if (imageBuffer.compare(NO_IMAGE_BUFFER) === 0) {
+    await wait(WAIT_MS) // try again after 500 ms
+    imageBuffer = await redis.getBuffer(fileName)
+  }
+  return imageBuffer
+}
+
+const ROUTE = `${JIRA_IMAGES_ENDPOINT}/`
+const jiraImagesHandler = uWSAsyncHandler(async (res: HttpResponse, req: HttpRequest) => {
+  const fileName = req.getUrl().slice(ROUTE.length)
+  const imageBuffer = await getImageFromCache(fileName)
+  if (imageBuffer.compare(NO_IMAGE_BUFFER) === 0) {
+    //TODO: send placeholder image
+  }
+
+  res
+    .writeStatus('200')
+    .writeHeader('Content-Type', 'image/png')
+    .writeHeader('Content-Length', imageBuffer.length.toString())
```

Comment:
@mattkrick after enabling proxy on dev server I was getting this
```
[HPM] Error occurred while trying to proxy request /jira-attachements/_redZ5TBVrI8sW7FgP79FTY-utdAdaAR_o0raZUkfXc from localhost:3000 to http://localhost:3001 (HPE_UNEXPECTED_CONTENT_LENGTH) (https://nodejs.org/api/errors.html#errors_common_system_errors)
```
I removed this and it works correctly. I'm not sure about prod env, is `Content-Length` going to be set automatically there too?

Reply:
here's what i'd try:
- remove the content-length header & on the browser open dev tools & check the response headers to verify that content-length is received. i think it should be
- if that doesn't work, i think you want byteLength, see https://nodejs.org/api/buffer.html#buffer_static_method_buffer_bytelength_string_encoding. try String(imageBuffer.byteLength)

Reply:
yeah, it is received. I just don't know if the prod env will behave the same as dev. I think it should 🤔

Reply:
oh for sure. uWS probably measures the payload before sending & attaches the necessary headers to keep the request from hanging, nginx just sits in front of it adding it's own headers so we're all good